iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/extensions/ActivePowerControlImpl.java

```diff
@@ -43,22 +49,75 @@ public ActivePowerControlImpl(T component,
                                   boolean participate,
                                   double droop,
                                   double participationFactor,
-                                  double minPOverride,
-                                  double maxPOverride) {
+                                  double minTargetP,
+                                  double maxTargetP) {
         super(component);
         int variantArraySize = getVariantManagerHolder().getVariantManager().getVariantArraySize();
         this.participate = new TBooleanArrayList(variantArraySize);
         this.droop = new TDoubleArrayList(variantArraySize);
         this.participationFactor = new TDoubleArrayList(variantArraySize);
-        this.minPOverride = new TDoubleArrayList(variantArraySize);
-        this.maxPOverride = new TDoubleArrayList(variantArraySize);
-        this.allTDoubleArrayLists = List.of(this.droop, this.participationFactor, this.minPOverride, this.maxPOverride);
+        this.minTargetP = new TDoubleArrayList(variantArraySize);
+        this.maxTargetP = new TDoubleArrayList(variantArraySize);
+        this.allTDoubleArrayLists = List.of(this.droop, this.participationFactor, this.minTargetP, this.maxTargetP);
+        double checkedMinTargetP = checkTargetPLimit(minTargetP, "minTargetP", component);
+        double checkedMaxTargetP = checkTargetPLimit(maxTargetP, "maxTargetP", component);
         for (int i = 0; i < variantArraySize; i++) {
             this.participate.add(participate);
             this.droop.add(droop);
             this.participationFactor.add(participationFactor);
-            this.minPOverride.add(minPOverride);
-            this.maxPOverride.add(maxPOverride);
+            this.minTargetP.add(checkedMinTargetP);
+            this.maxTargetP.add(checkedMaxTargetP);
+        }
+        checkLimitOrder(minTargetP, maxTargetP);
+    }
+
+    record PLimits(double minP, double maxP) { }
+
+    private PLimits getPLimits(T injection) {
+        double maxP = Double.MAX_VALUE;
+        double minP = -Double.MAX_VALUE;
+        if (injection instanceof Generator generator) {
+            maxP = generator.getMaxP();
+            minP = generator.getMinP();
+        } else if (injection instanceof Battery battery) {
+            maxP = battery.getMaxP();
+            minP = battery.getMinP();
+        }
+        return new PLimits(minP, maxP);
+    }
+
+    private double withinPMinMax(double value, T injection) {
+        PLimits pLimits = getPLimits(injection);
+
+        // The test always returns false if value is undefined (Double.Nan) as it should,
+        // as side effect of Nan arithmetic operators Nan handling, as they are used below.
+        if (value < pLimits.minP || value > pLimits.maxP) {
+            LOGGER.warn(String.
+                    format("targetP limit is now outside of pMin,pMax for component %s. Returning closest value in [pmin,pMax].",
+                            injection.getId()));
```

Comment:
The right way to fix the Sonar issue is to use SLF4J parameterized logging. Its purpose is to improve the performance.
You can read more about this [here](https://www.slf4j.org/faq.html#logging_performance).

```suggestion
            LOGGER.warn("targetP limit is now outside of pMin,pMax for component {}}. Returning closest value in [pmin,pMax].",
                            injection.getId());
```

Reply:
Done. That makes more sense. Thanks for the explanation.